internal/animation/runner.go

```diff
@@ -47,19 +69,22 @@ func (r *Runner) runAnimations() {
 			<-draw.C
 			r.animationMutex.Lock()
 			oldList := r.animations
-			r.animations = nil // clear the list so we can append any new ones after processing
 			r.animationMutex.Unlock()
-			var newList []*anim
+			newList := make([]*anim, 0, len(oldList))
 			for _, a := range oldList {
-				if r.tickAnimation(a) {
+				if r.tickAnimation(a) && !a.isStopped() {
```

Comment:
Shouldn't we check if it is stopped before ticking it? (i.e. these arguments the other way round).

Reply:
Hmm I don't think so, the idea of writing it after the tick is that if inside the tick, the animation is stopped, we don't append it to the new list.

Reply:
But as animations will more commonly be stopped from outside is it not sensible to avoid the tick? If stop is set during tick it will still be cancelled, just before the next tick rather than after the current?

Reply:
Yes, you are right, we can move it before or added before. I mean:
```go
if !a.isStopped() && r.tickAnimation(a) && !a.isStopped() {
```
Then animations are always stopped immediately and `isStopped` is not an expensive function, is it?

Reply:
I guess it's not expensive, but checking in both places seems like it could be overkill as we will find out in the next iteration anyway.
I'd be wary of calling mutexes in tight loops if it can be avoided - but maybe you know more about the specifics herer and how it performs.

Reply:
Yeah, you are right, I am just worried about appending stopped animations in a slice will increase its capacity (that it is not needed). But, maybe that's something we can leave as it and check if we have problems later. I will update :)

Reply:
done